### DISCUSSION TITLE
Support PDF embed and additional file attachment

### DISCUSSION THREAD

**Discussion by hsingll** [alangrainger/share-note#119] [Vote for upcoming features]
Only framwork displays with no content for now.
<img width="743" alt="Snipaste_2024-08-02_10-31-51" src="https://github.com/user-attachments/assets/a1d8093e-8a4c-405f-a65d-2760f5715838">

**Comment by alangrainger** [comment]
Hi @hsingll , thanks for the suggestion but it is out of scope for this plugin.

Share Note is for sharing your notes; it's not a file-sharing service.

The correct way to share a PDF or any large file is to upload it to a proper file sharing service, and then link the document back into your note.

See also https://github.com/alangrainger/share-note/discussions/38

And https://docs.note.sx/faq#will-you-add-support-for-pdfs

**Reply by karmaniverous** [reply]
Hey @alangrainger I think I might be speaking for a lot of Evernote refugees when I say that sharing a PDF in a note, either as an embed or as a link, is actually a key use case. There are just SO MANY ways that documents wind up in the pile as PDFs... paper scans, printing emails to PDF, and on and on.

The "correct way" you cite above is just a non-starter. I use a note app like Evernote (and now Obsidian) precisely because I DON'T want the extra friction of monkeying around with a file-sharing service. I want to share a note ad hoc and have the resulting page just WORK.

What do you think it would take to get this working? Because with the fall of Evernote I think the business case may be there... I for one would be DELIGHTED to pay a monthly fee for this functionality.

Would you be willing to put a little time into gaming it out with me?

**Reply by alangrainger** [reply]
I like the way you frame this @karmaniverous  - I am also an Evernote refugee. I will have a think about how achievable this is to implement.

**Reply by karmaniverous** [reply]
Awesome sauce! Happy to help if you want to kick the ball around, DM me at https://x.com/karmaniverous

**Reply by nicmart-dev** [reply]
Any progress on this? PDF support is really a must to make notes sharing functional (eg. I'm sharing a travel itinerary of our holiday to my wife and she can't access embedded pdf with flight booking confirmation). Or is there another plugin people recommend instead, that is just as convenient 1 click share to use as this one?

As to providing a solution for this plugin, perhaps serving files off a selfhosted https://www.min.io/
